src/index.js

```diff
@@ -20,37 +20,43 @@ const uploadBundle = require("./upload");
  * Generates an entrypoint for bundling the handlers
  * It also makes sure all handlers are registered with the runtime
  *
- * @param {string} src path to the edge handler directory
+ * @param {string} sourceDir path to the edge handler directory
  * @returns {Promise<{ handlers: string[], mainFile: string }>} list of handlers and path to entrypoint
  */
-async function assemble(src) {
-  const tmpDir = await fsPromises.mkdtemp(path.join(os.tmpdir(), "handlers-")); //make temp dir `handlers-abc123`
-  const handlers = [];
-  const imports = [];
-  const registration = [];
-  const functions = await fsPromises.readdir(src, { withFileTypes: true });
+async function assemble(sourceDir) {
+  const entries = await fsPromises.readdir(sourceDir, { withFileTypes: true });
+  const handlers = entries.filter(isHandlerFile).map(getFilename);
 
-  for (const func of functions) {
-    const file = path.parse(func.name);
-
-    if (!func.isFile() || (file.ext !== ".js" && file.ext !== ".ts")) {
-      continue;
-    }
+  if (handlers.length === 0) {
+    return { handlers };
+  }
 
+  const imports = [];
+  const registration = [];
+  for (const handler of handlers) {
     const id = "func" + crypto.randomBytes(16).toString("hex");
 
-    imports.push(`import * as ${id} from "${path.resolve(src, file.name)}";`);
-    registration.push(`netlifyRegistry.set("${file.name}", ${id});`);
-    handlers.push(file.name);
+    imports.push(`import * as ${id} from "${path.resolve(sourceDir, handler)}";`);
+    registration.push(`netlifyRegistry.set("${handler}", ${id});`);
   }
-
-  // import path //
   const mainContents = imports.concat(registration).join("\n");
+
+  const tmpDir = await fsPromises.mkdtemp(path.join(os.tmpdir(), "handlers-")); //make temp dir `handlers-abc123`
   const mainFile = path.join(tmpDir, MAIN_FILE);
   await fsPromises.writeFile(mainFile, mainContents);
   return { handlers, mainFile };
 }
 
+function isHandlerFile(entry) {
+  return entry.isFile() && ALLOWED_EXTENSIONS.includes(path.extname(entry.name));
+}
+
+const ALLOWED_EXTENSIONS = [".js", ".ts"];
```

Comment:
Since we don't transpile ts files with this plugin, shouldn't we only allow .js extensions?

Maybe we can provide an error message asking folks to transpile their code if we come across .ts extensions?

Reply:
Good point! Fixed.